fendermint/app/src/settings.rs

```diff
@@ -58,7 +59,21 @@ pub struct FvmSettings {
 pub struct IPCSettings {
     #[serde(deserialize_with = "deserialize_subnet_id")]
     pub subnet_id: SubnetID,
-    pub config: fendermint_vm_topdown::Config,
+    /// The config for top down checkpoint. It's None if subnet id is root or not activating
+    /// any top down checkpoint related operations
+    pub topdown: Option<fendermint_vm_topdown::Config>,
+}
+
+impl IPCSettings {
+    pub fn is_topdown_enabled(&self) -> bool {
+        !self.subnet_id.is_root() && self.topdown.is_some()
+    }
+
+    pub fn top_down_config(&self) -> anyhow::Result<&fendermint_vm_topdown::Config> {
```

Comment:
```suggestion
    pub fn topdown_config(&self) -> anyhow::Result<&fendermint_vm_topdown::Config> {
```

Either use `top_down` everywhere or `topdown` but let's not mix them.